File: examples/csharp/language_gauntlet/Runtime/Executor.cs
namespace LanguageGauntlet.Runtime;

using System.Diagnostics;

public static class Executor
{
    public static string Execute(string cmd)
    {
        // SINK -- Process.Start / CWE-78.
        Process.Start("cmd.exe", "/c " + cmd);
        return cmd;
    }

    public static string CleanTwin()
    {
        // NEGATIVE -- the constant argument must remain untainted.
        Process.Start("cmd.exe", "/c echo clean");
        return "clean";
    }
}

File: examples/c/language_gauntlet/src/runtime/executor.c
#include <stdlib.h>

#include "gauntlet/executor.h"

int execute(const char *cmd) {
    /* SINK -- system() / CWE-78. */
    return system(cmd);
}

int clean_twin(void) {
    /* NEGATIVE -- the constant argument must remain untainted. */
    return system("echo clean");
}
